Switch to vector images

isiko opened the pull request:
# Description
* Replace old PNG-Images with SVG Images
* Replace Custom Images with Material Icons
* Recreate old PNG-Images as SVG Images

# Done
- [x] Update related screenshots in [customization guide](https://github.com/EventFahrplan/EventFahrplan/blob/master/docs/CUSTOMIZING.md).
- [x] Check if `app:srcCompat` is needed for older Android versions, see `minSdk`.
- [x] Check if `vectorDrawables.useSupportLibrary = true` is needed for older Android versions, see `minSdk`.

# Screenshots
| Screen | Before | After |
| --------- | --------- | ------ |
| Main Screen | ![Mainscreen-Before](https://user-images.githubusercontent.com/44641688/194672904-3c093a77-3e9d-4e35-a019-fed12e074113.png) ![image](https://user-images.githubusercontent.com/44641688/194750732-c8b1ba39-b554-4978-8b1d-aa303b0d1c2f.png) | ![Mainscreen-After](https://user-images.githubusercontent.com/44641688/194672898-8522da89-4775-43f4-8cc7-dfebc83f9e5e.png) ![image](https://user-images.githubusercontent.com/44641688/194750688-14addd4e-cd19-422c-8d96-b9c0ae2660e1.png) |
| Session Screen | ![Session-Before](https://user-images.githubusercontent.com/44641688/194672957-ec4c0f05-882e-4977-9be5-5edcf795ff02.png) | ![Session-After](https://user-images.githubusercontent.com/44641688/194673387-a03f7fcd-e22d-4ce2-b402-4e64525619e2.png) | 
| Change Screen | ![Change-Before](https://user-images.githubusercontent.com/44641688/194705976-3202c595-2f78-42b3-a5f0-47ce491d2429.png) | ![After](https://user-images.githubusercontent.com/44641688/195211846-8ea36f02-9e25-423a-99b4-d9fa47e65bfe.png) |

# Tested on (with all build flavors):
- Phone (Pixel 2), API lvl 24 Emulator
- Phone (Pixel 2), API lvl 33 Emulator

---

Resolves #510
Resolves #484

isiko commented:
Renamed some files, should be good now

johnjohndoe commented on app/src/main/res/menu/starred_list_menu.xml:
```diff
@@ -36,5 +36,5 @@
             app:showAsAction="ifRoom"
-            android:icon="@drawable/ic_delete_white_24dp"/>
+            android:icon="@drawable/ic_delete"/>
 </menu>
```
In assets, you can simplify the files names like:

- assets/bell_list.svg
- assets/healing_list.svg
- assets/star_list.svg

johnjohndoe commented on app/src/main/res/menu/starred_list_context_menu.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
```
:information_source: Please list the generic icon names which are changed in the commit message body so it easier to search for them and get a quick overview.

johnjohndoe commented on app/src/main/res/menu/alarms_menu.xml:
```diff
@@ -3,7 +3,7 @@
 
```
:warning: Notification icons are specific to the build flavor. Please do not delete them.

isiko replied:
I know, but there should already be a Vector Version of that Image :thinking:

johnjohndoe replied:
If I understand you correctly you are referring to [drawable-anydpi-v24](https://github.com/EventFahrplan/EventFahrplan/tree/master/app/src/ccc36c3/res/drawable-anydpi-v24). The `v24` suffix of the folder looks like the vector drawable might only be supported as of SDK level 24. For now the icon assistant in Android Studio still generates PNGs, too. I guess there is a reason for it. But if you find out that PNGs are no longer necessary for old and new Android versions that would be beautiful.

johnjohndoe replied:
Please note the general (not specific to notification icons) SDK level hints on these pages:

- https://developer.android.com/develop/ui/views/graphics/vector-drawable-resources
- https://developer.android.com/develop/ui/views/graphics/drawables
- https://developer.android.com/reference/android/graphics/drawable/VectorDrawable
- https://developer.android.com/jetpack/androidx/releases/vectordrawable/

There is also a couple of videos from Google Developers and conferences about the topic.

isiko replied:
Also found a [thread on Stackoverflow](https://stackoverflow.com/questions/34016107/can-we-use-vectordrawable-or-vectorxml-as-icons-for-push-notifications-in-androi) about that topic now, I'll add those images back in :+1:

johnjohndoe commented on app/src/main/res/drawable/ic_star_full.xml:
```diff
@@ -0,0 +1,8 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
:thinking: Commit message: Why `Replace Reminder Icons`? How about `Replace icons to un/favor a session.`?

Side note: you do not have to capitalize most of the English words in your commit messages. Also have read [here](https://cbea.ms/git-commit/).

johnjohndoe reviewed (changes requested):
# First review
- Thank you for taking the effort of replacing pixel icons with vector files. :+1: 
- Please provide before and after **screenshots** of the relevant screen parts as suggested in the [contributing guide](https://github.com/EventFahrplan/EventFahrplan/blob/master/CONTRIBUTING.md).
- The icons you added appear much smaller on the device (Pixel 6) I tested with. Please try to match the former dimensions. I find them **too small** to recognize the details.
- The **multi-color icons** (camera) need a bit more thinking. I could imagine that it is possible to create a layered icon where the red bar is one layer and the filled camera icon is a layer below which can be tinted. [`layer-list`](https://developer.android.com/guide/topics/resources/drawable-resource#LayerList) might be worth a try. If the camera topic is to difficult to solve you can leave the former PNG icons and we update them in a separate pull request.
- Please check on older Android version if `app:src` works with the vector drawables and document in the pull request which versions you tested with. EDIT: typo, I meant `android:src`.

isiko commented:
>     * Please provide before and after **screenshots** of the relevant screen parts as suggested in the [contributing guide](https://github.com/EventFahrplan/EventFahrplan/blob/master/CONTRIBUTING.md).

Added Screenshots (The *after* already have some changes to the Size of the Icons that I haven't pushed yet)
 
>     * The icons you added appear much smaller on the device (Pixel 6) I tested with. Please try to match the former dimensions. I find them **too small** to recognize the details.

I am seemingly not able to match it 100%, but it should be better now (as seen in the Screenshot's I added)

>     * The **multi-color icons** (camera) need a bit more thinking. I could imagine that it is possible to create a layered icon where the red bar is one layer and the filled camera icon is a layer below which can be tinted. [`layer-list`](https://developer.android.com/guide/topics/resources/drawable-resource#LayerList) might be worth a try. If the camera topic is to difficult to solve you can leave the former PNG icons and we update them in a separate pull request.

I'll look into that as soon as possible :+1: 

>     * Please check on older Android version if `app:src` works with the vector drawables and document in the pull request which versions you tested with.

Sorry, forgot to put that in the PR, adding that now :facepalm:

isiko commented on app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt:
```diff
@@ -118,7 +118,10 @@ abstract class SessionsAdapter protected constructor(
-        viewHolder.withoutVideoRecording.setImageResource(R.drawable.ic_without_video_recording)
+        viewHolder.withoutVideoRecording.apply {
+            setImageResource(R.drawable.ic_novideo)
+            setColorFilter(R.color.session_list_item_no_video)
```
I tried to use one color per Icon for the Video Camera Icons, making them White, Grey and Red, but I will replace that with the original colors and the red bar from the old Icons :+1:

isiko commented on app/src/main/res/values/colors.xml:
```diff
@@ -57,6 +58,10 @@
     <color name="session_list_empty_text">#000000</color>
+    <color name="session_list_item_video">@color/colorAccent</color>
+    <color name="session_list_item_no_video">#FF0000</color>
+    <color name="session_list_item_without_video_recording">#656565</color>
```
The `session_list_item_without_video_recording` color is a left over, but the others should be used by the Video Camera Icon ( although I will rework those )

isiko commented on app/src/main/res/layout/session_list_item.xml:
```diff
@@ -57,7 +58,8 @@
         android:contentDescription="@string/session_list_item_video_content_description"
-        android:src="@drawable/ic_change_video"
+        android:src="@drawable/ic_video"
+        android:tint="@color/session_list_item_video"
```
Thanks for that referance, I actually had a really hard time understanding the different usages of the Icon, I'll use this as the base for the new version :+1:

johnjohndoe replied:
You are always welcome to ask me when you are stuck. Did you see the usage of `ic_without_video_recording_took_place` here?

https://github.com/EventFahrplan/EventFahrplan/blob/master/app/src/main/java/nerd/tuxmobil/fahrplan/congress/favorites/StarredListAdapter.kt#L74

johnjohndoe replied:
I must admit I am not too happy with the aesthetics of the new "blocky" camera icon. I looked around and found the following one:

- http://simpleicon.com/wp-content/uploads/video-camera-7.svg

Can you try how it looks with a red bar?

isiko replied:
This would be the Black Version with a Red Bar:
![video-camera-withBar](https://user-images.githubusercontent.com/44641688/194715037-e299b21e-bb6d-4369-a85a-9f3a6673fe23.svg)

johnjohndoe replied:
That looks really good. Do you like it, too?

isiko replied:
I'm not 100% sold on the Icon as I woul prefer a bit more simplistic Icon, but I wouldn't complain about it 👍

isiko replied:
Whoops, that should not have happend. I fixed that and double checked all Camera Icons, so they should now all be correct in color, size and having a bar or not

johnjohndoe commented on app/src/main/res/drawable/ic_action_star_list.xml:
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
_Reposting the comment in a resolvable thread_

>> Please provide before and after screenshots of the relevant screen parts as suggested in the [contributing guide](https://github.com/EventFahrplan/EventFahrplan/blob/master/CONTRIBUTING.md).

> Added Screenshots (The after already have some changes to the Size of the Icons that I haven't pushed yet)

Please note the other camera icon usages in the main screen and in the favorites and changes screens.

johnjohndoe replied:
If you do the screenshots in landscape mode then the schedule changes icon also is visible. You can cut out that part - no need to show the whole screen.

johnjohndoe replied:
Thank you for the additional screenshot. - Please update the outdated to reflect the current state of this PR.

isiko replied:
Done :+1:

johnjohndoe replied:
Can you please try a Pixel 6 emulator? The icons look way smaller there for me.

isiko replied:
Do you mean all Icons? For me at least, I think most icons have a similar size, except for the List Icons, where the Star/Bell/Bandaid seem to be quite small

johnjohndoe replied:
For me, also the camera icons are smaller on the main screen and in the favorites screen. Likely also in the schedule changes screen, I did not check.

isiko replied:
I looked at the Screenshots and could not see that big of a difference, for the List Icons it seems that the Smaller Icons were made a bit bigger, which I could try to recreate, for the Camera Icon it might be that the old Icon was filled out entirely while the new Icon has a lot more detail, showing more of the background, which might make it look smaller?

johnjohndoe replied:
Here is how the "no video recording" icon looks like on a Pixel 6 device (old / new):

![no-video-old-new-pixel-6](https://user-images.githubusercontent.com/144518/197629276-cfb250f3-6266-4e2a-aed1-a32e1691aaf6.png)

isiko replied:
Should be fixed in next push

johnjohndoe replied:
I will check on my device. Why did you also change the XML settings for the bell icon now? Did they also show up small?

johnjohndoe replied:
:heavy_check_mark: Checked here: https://github.com/EventFahrplan/EventFahrplan/pull/518#discussion_r1082605912

johnjohndoe commented on app/src/main/res/drawable/ic_action_star_list.xml:
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
_Reposting the comment in a resolvable thread_

>> The icons you added appear much smaller on the device (Pixel 6) I tested with. Please try to match the former dimensions. I find them too small to recognize the details.

> I am seemingly not able to match it 100%, but it should be better now (as seen in the Screenshot's I added)

What blocks you here from matching the size?

isiko replied:
I think it's because of the switch from handmade Icons to icons provided by Google. I can get them to a difference of a few pixels, but I think the base Vector Image has ever so slightly different properties so I would have to spend a lot of time fine-tuning that.

johnjohndoe replied:
I can hopefully help with that. The icons are still too small to be recognizable very well. I try to find out how to fix it.

johnjohndoe replied:
I made some progress on a [separate branch](https://github.com/EventFahrplan/EventFahrplan/tree/switchToVectorImages_refinement). There are indeed some fine-tuning steps which need time and patience. I will continue there once I have time. :hourglass_flowing_sand:

johnjohndoe replied:
Here is the current state of the icons comparing `master` (d6480389f87df94f499fb7a9bd26f49f808b99e1) with the state of the separate branch (9bfe7aa2307f163584e3334fbb9a665ef9625e80):

# Schedule screen (before / after)
- Icons in the toolbar
- Camera icon on a session view
![20230120_schedule_screen_tablet](https://user-images.githubusercontent.com/144518/213721153-980b6e70-2cb6-416e-a9b2-83c3ec029585.png)

# Favorites screen (before / after)
- Camera icon on a past session view
- Camera icon on a future session view
![20230120_favorites_screen_tablet](https://user-images.githubusercontent.com/144518/213721182-c4880b7e-3a9a-4bfd-88bf-d697f6ca9a6f.png)

These look good to me. Any objections?

isiko replied:
Nope, looks good to me :)

isiko commented:
I've now implemented most of the requested Changes and redid the Video Icons. They currently are not multi-colored, but at least I think I got the colors right now, I'll have a look into making that bar red now.

johnjohndoe commented:
Here is two website which might be useful for the icons creation:

- https://material.io/design/iconography/system-icons.html#icon-themes (does and don'ts
- https://fonts.google.com/icons?icon.query=camera&icon.style=Rounded&icon.platform=android (online generator)

johnjohndoe commented on assets/video-camera-withBar.svg:
```diff
@@ -0,0 +1,43 @@
+   id="svg38"
+   sodipodi:docname="video-camera-7.svg"
+   xml:space="preserve"
+   inkscape:version="1.2.1 (0f2f062aeb, 2022-09-21, custom)"
```
Please save this file as `Plain SVG (*.svg)` (a file format option in Inkscape) instead of `Inkscape SVG (*.svg)`. In this case we don't need the metadata.

Please use lowercase for the file name to match the others: `video-camera-with-bar.svg`

johnjohndoe replied:
Thanks. No my branch does not touch the video camera icons.

johnjohndoe commented on app/src/main/res/drawable/ic_change_novideo.xml:
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
In the latest rebase you are missing to remove the following files in your last commit. Before the rebase you removed the `rc3` flavor files which was fine. Now you can remove the same `jev2022` flavor files (`ic_change_novideo.png`, `ic_change_video.png`, `ic_novideo.png`, `ic_without_video_recording.png`).

isiko replied:
Done in next Push

isiko commented on assets/video-camera.svg:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Generator: Adobe Illustrator 16.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
Should be done with next Push, allthough I think this will be overridden with your improvements anyway :D

johnjohndoe replied:
Thanks. No my branch does not touch the video camera icons.

isiko commented:
Sorry for the double push, something went wrong with rebasing

johnjohndoe commented on app/src/main/res/layout/session_layout_land_large.xml:
```diff
@@ -37,7 +37,7 @@
             android:layout_marginLeft="5dp"
             android:adjustViewBounds="true"
             android:baseline="14sp"
-            android:src="@drawable/bell" />
```
I think there is left-over files here:

- app/src/cccamp2019/res/drawable/bell.png
- app/src/jev2022/res/drawable/bell.png

johnjohndoe commented on app/src/main/res/drawable/ic_change_video.xml:
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
There a PNG still here:

- app/src/cccamp2019/res/drawable/ic_change_video.png

johnjohndoe commented on app/src/main/res/layout/session_list_item.xml:
```diff
@@ -67,7 +67,7 @@
     <ImageView
         android:id="@+id/session_list_item_without_video_recording_view"
         android:layout_width="wrap_content"
-        android:layout_height="14sp"
```
Shouldn't the changes to video recording icons be in the "Updated Video Icons" commit?

isiko replied:
The Problem wasn't just with the Video Icons but also the Bell Icon on the Main Screen, so there are also [changes](app/src/main/res/layout/session_layout_land.xml) regarding that. I'll move the Changes to 9c5be3a073b4ff888c442e4c24b6c4dfd34a62f5

isiko replied:
As I don't want to pull those changes out of a commit again and didn't have confirmation from you yet I've put them in a separate commit for now and will squash that into whichever commit you find suitable

johnjohndoe replied:
The layout for the bell icon on the schedule screen seems to be a bit tricky. In the current implementation of your branch the bell icon sometimes gets **cut off**. Here is a few examples (before/after):

![20230120_schedule_screen_alarm_bells_tablet](https://user-images.githubusercontent.com/144518/213722394-ab843254-0d9b-4eef-8db1-2dc633b6563b.png)

I believe this is why formerly the size of the icon was chosen to **depend on the font size**, hence `sp` instead of `dp`. It might be a general deficit of **how the layout** has been done. Do you want to digg into this issue? The alternative would be to extract the changes concerning the bell icon on the session screen from this pull request and handle it **separately**. All other changes seem to be ready to merge as far as I can tell.

isiko replied:
I don't think it would be possible to extract that changes, as then the Bell would have the same Size-Problem as the Video Icon. I'll look into this issue :+1:

johnjohndoe replied:
:+1: Please do. I am not sure if both are affected since they are defined in different layout files.

johnjohndoe commented on app/src/main/res/layout/session_list_item.xml:
```diff
@@ -76,7 +76,7 @@
         android:baseline="13sp"
         android:contentDescription="@string/session_list_item_without_video_content_description"
-        android:src="@drawable/ic_without_video_recording"
+        android:src="@drawable/ic_novideo"
```
# Schedule screen: camera icon
I only now noticed a **little vertical offset** between the title text and the camera icon. You can see it pretty well when you enable layout bounds in the developer settings.

![20230120_schedule_screen_camera_icon_phone](https://user-images.githubusercontent.com/144518/213742877-f72d5928-a02b-4b14-8f2e-6ac50d71383a.png)

Can we please fix this?

isiko replied:
This should be fixed now, although I couldn't find the option you mentioned

johnjohndoe replied:
Go into "Settings" -> "System" -> "Developer Options" (must be activated before) and there search for "Show layout bounds".

isiko replied:
Thanks, I actually forgot to update this for Tablets, will be fully fixed in next push

johnjohndoe replied:
Can we position the icons in the **top right** corner matching the **first row** of the single or multiline text? 

I also noticed that the camera and bell icons appear **smaller** for a single line title and **bigger** for a multiline title. Can we always use the same size? This can be seen in the 36C3 app in the "Chaos-West Bühne" column on a phone in landscape mode.

![icons1](https://user-images.githubusercontent.com/144518/214436352-45b51c85-4542-4ef1-88b9-6e0772b764f7.png) ![icons2](https://user-images.githubusercontent.com/144518/214436356-6a9e6d5d-c074-404c-8ba4-d9e10d569368.png)

isiko replied:
Always using the same size is no Problem, using the Top Right also isn't a problem, the only weird thing is the size. I think I've got the correct size now, but somehow if I use the defined Text Size (16sp) as the height for the Icons, they seem a lot smaller. I am currently using about 1.375x the text size which seems to be the correct height, but I find this very strange. It's probably some padding added to the text, but I don't see where that would be defined :(

johnjohndoe replied:
In commit 48124f565735f188980bf5243877de1119ed29ee the size difference is still present in landscape mode. For you, too?

isiko replied:
Which Device / Screen?

isiko replied:
Fixed

johnjohndoe replied:
I used a Pixel 6, Android 13 in landscape mode.

isiko replied:
Nice, I also tested on that, should be fixed now :+1:

johnjohndoe replied:
Looks good on my Pixel 6. Can you please document why the `layout_height` values are needed resp. how you found them? Also I would be interested what `fitEnd` vs `fitStart` changes for the icons.

isiko replied:
I can add comments that the Values values were found by trial and error and that they generally are 1,725x the old value for the old grid-based Values. Probably something like this: 

```
<!-- Scale adjusted by 1.725x to match font size while switching to Vector Graphics (Value found by Trial and Error) -->
```

`fitStart` makes the Icons render at the start of their Bounding box if they don't completely fill it (in this case the relevant thing is that they render at the top), while `fitEnd` makes them render at the End of said box. I hope you didn't find any instances of me using `fitEnd`? If so, that probably wasn't intentional

johnjohndoe replied:
The commit message details is a good place for the explanation.

isiko replied:
Added :+1:

johnjohndoe commented on app/src/main/res/layout/session_list_item.xml:
```diff
@@ -67,14 +65,13 @@
         android:id="@+id/session_list_item_without_video_recording_view"
         android:layout_width="wrap_content"
-        android:layout_height="14sp"
+        android:layout_height="wrap_content"
```
Why is this icon set to `wrap_content` while the others are set to `24dp`?

isiko replied:
Just forgott to change that :sweat_smile:

isiko replied:
I realized that I actually should have changed the others to, so now they all are at `24.15sp`, so they scale with the font size :+1:

johnjohndoe commented on app/src/main/res/layout-port/session_layout.xml:
```diff
@@ -24,22 +24,18 @@
             android:layout_width="wrap_content"
-            android:layout_height="13sp"
-            android:layout_marginLeft="3dp"
+            android:layout_height="22sp"
```
The bell icon adjusts to the height (without the red bar) of the camera icon next to it by adding `android:padding="3dp"` here. I think this looks a bit better.

:heavy_check_mark: Tested on:

- Pixel 6, Android 13, portrait and landscape modes
- 10" tablet, Android 13 in landscape and landscape split mode.

johnjohndoe commented on app/src/main/res/layout/session_layout_land.xml:
```diff
@@ -24,22 +24,18 @@
             android:layout_width="wrap_content"
-            android:layout_height="9sp"
-            android:layout_marginLeft="3dp"
+            android:layout_height="16.5sp"
```
The bell icon adjusts to the height (without the red bar) of the camera icon next to it by adding `android:padding="2dp"` here. I think this looks a bit better.

:heavy_check_mark: Tested on:

- Pixel 6, Android 13, portrait and landscape modes
- 10" tablet, Android 13 in landscape and landscape split mode.

johnjohndoe commented on app/src/main/res/layout/session_layout_land_large.xml:
```diff
@@ -24,20 +24,18 @@
             android:layout_width="wrap_content"
-            android:layout_height="16sp"
-            android:layout_marginLeft="5dp"
+            android:layout_height="27.5sp"
```
The bell icon adjusts to the height (without the red bar) of the camera icon next to it by adding `android:padding="4dp"` here. I think this looks a bit better.

:heavy_check_mark: Tested on:

- Pixel 6, Android 13, portrait and landscape modes
- 10" tablet, Android 13 in landscape and landscape split mode.

johnjohndoe reviewed (changes requested):
I played around with the size and positioning of the bell icon(s) in the schedule screen somewhat. Here is what I came up with to visually align the height of the bell and camera icons.

johnjohndoe commented:
I now merged your changes with mine and pushed the latest state.
I also checked the app on API level 21 and documented the findings.
Here are screenshots of the current state:

![events](https://user-images.githubusercontent.com/144518/218338680-fb30c6c9-e5a7-48b7-9c97-b272538d8a29.png)
![events-landscape](https://user-images.githubusercontent.com/144518/218338679-c6756825-a61b-4bed-b788-f30da6430ebb.png)
![event](https://user-images.githubusercontent.com/144518/218338681-010e1890-aecf-4b04-92d7-d41be1cfc81d.png)
![favorites](https://user-images.githubusercontent.com/144518/218338678-fdbf5327-b23f-4207-b4c1-998b48016cf2.png)

johnjohndoe reviewed (approved):
Ready to merge! :rocket:

Thank you for your patience with me and this long-standing pull request. Thank you for working on this!!!
